### PR TITLE
Piggy extractor prototype (distributed IO-free pipe-based language-agnostic extractors)

### PR DESCRIPTION
**Pull request by alldefector**
examples/spouse_example/piggy_extractor/application.conf works with PG
and PGXL. Haven’t tested on GP.

### CODE REVIEW

File: doc/doc/basics/extractors.md [hunk 1]
```diff
@@ -17,10 +17,15 @@ class contains different extractor *styles*:
 
 - Row-wise extractors:
   - [`json_extractor`](#json_extractor): highly flexible and compatible with
-  previous systems, but with limited performance 
+  previous systems, but with limited performance
```
Comment (netj) [inline doc/doc/basics/extractors.md]: Could you please keep the trailing whitespace cleanup in a separate PR?  It's really hard to see what's really changed.  They should've not introduced in the first place, but I think it's even worse if we clean them up in random commits.

Comment (alldefector) [inline doc/doc/basics/extractors.md]: Last time I tried reverting space changes, I ended up destroying my working copy...

Good news: Adding 'w=1' to the URL suppresses whitespace differences: https://github.com/tiimgreen/github-cheat-sheet#ignore-whitespace

Also, if we switch to Phabricator, we get smarter diff by default, together with lots of configurable linters.

Comment (netj) [inline doc/doc/basics/extractors.md]: I'm worried because the whitespace confuses git.  Yes, we humans can use brain or tools to figure out what's different, but once such commit mixed with a lot of unrelated changes is put into history, it's really hard to backport, merge with other branches, etc.  Sorry, but it was a big pain to merge conflicts created by some of your past commits again and again: it's really a disaster for integrators and release engineers.  If your editor is stripping all the trailing whitespaces, I would ask you to disable that setting temporarily for DeepDive until we do a full cleanup on every file and add a style check to ban them forever :)  For this one I'll try to merge as it is right after the release, hopefully without much trouble.

Comment (alldefector) [inline doc/doc/basics/extractors.md]: I see. Sorry about the trouble... I'll rebase after the release and handle the conflicts from my side so that this change stays automatically mergeable.

If the conflicts you mentioned were b/w master and develop, I think the real solution is to have one conceptual "master" branch and avoid changes flying in both directions (colliding with each other).

Comment (netj) [inline doc/doc/basics/extractors.md]: That'd be great!  The conflicts were mainly between ones trying to get into develop.  Well, master and develop is another thing we need to iron out, and we'll have a true "master" after this release with a clear doc on the branching/release policy.

Comment (alldefector) [inline doc/doc/basics/extractors.md]: Start adopting Phabricator today, and it'll become the contributor's responsibility (as opposed to the maintainer's) to handle conflicts (rebase etc.)! :)

Comment (chrismre) [inline doc/doc/basics/extractors.md]: Feng, with strong opinions about tools? At least I have no strong opinions
:)

Jaeho is the boss :)

Comment (alldefector) [inline doc/doc/basics/extractors.md]: I was mostly joking :) And github isn't bad at all! I was just really impressed by FB's tool that I couldn't help but to peddle it all the time :)
